### PR TITLE
[popover] Disable focus management when opened via hover

### PR DESCRIPTION
**Pull request by atomiks**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/react/src/Popover/Root/usePopoverRoot.ts [hunk 1]
```diff
@@ -69,22 +68,21 @@ export function usePopoverRoot(params: usePopoverRoot.Parameters): usePopoverRoo
     (nextOpen: boolean, event?: Event, reason?: OpenChangeReason) => {
       onOpenChange(nextOpen, event, reason);
       setOpenUnwrapped(nextOpen);
-      if (!keepMounted && !nextOpen) {
```
Comment (atomiks) [inline packages/react/src/Popover/Root/usePopoverRoot.ts:72]: Need to check how `setMounted(false)` with this `keepMounted` check operates now for the other components with the `hidden` prop

Comment (atomiks) [reply packages/react/src/Popover/Root/usePopoverRoot.ts:72]: Looks like `keepMounted` was never passed to the hooks, hence it worked like `keepMounted=false`. I've removed the prop for Popover but it should be done for all components.

**Comment by michaldudak** [comment]
It feels wrong that the popover closes on mouseout after an input inside it has been focused. Clicking on anything inside the popover should make it sticky

**Comment by atomiks** [comment]
@michaldudak that's a separate discussion as the previous focus management also worked like this

**Review by michaldudak — APPROVED** [review]

**Pull request merged**